### PR TITLE
docs: cross-reference sigmoid APIs

### LINKED ISSUE #105318: [docs] torch.sigmoid to make clear equivalence relations to other sigmoid functions
### 📚 The doc issue

Currently https://pytorch.org/docs/stable/generated/torch.sigmoid.html only points to https://pytorch.org/docs/stable/special.html?highlight=expit#torch.special.expit

Both do not link to https://pytorch.org/docs/stable/generated/torch.nn.Sigmoid.html?highlight=sigmoid#torch.nn.Sigmoid or https://pytorch.org/docs/stable/generated/torch.Tensor.sigmoid.html?highlight=sigmoid#torch.Tensor.sigmoid

https://pytorch.org/docs/stable/special.html?highlight=expit#torch.special.expit (and other special functions) should deserve their own `.html` pages btw (as all other functions). expit should mention explicitly if it's just an alias to torch.sigmoid and torch.nn.functional.sigmoid

Ideally all these function variants should just say "See [some single variant]"

Currently there exists 4 functional variants and 1 module variant and findable in docs:
- torch.sigmoid
- torch.Tensor.sigmoid
- torch.nn.functional.sigmoid
- torch.special.expit
- torch.nn.Sigmoid

As usual, search results also contain duplicates and snippets are quite bad :(
image

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by deepujain**
Fixes #105318

## Summary
- Make `torch.nn.Sigmoid` the central docstring that names the equivalent sigmoid APIs.
- Link `torch.sigmoid`, `Tensor.sigmoid`, `torch.nn.functional.sigmoid`, and `torch.special.expit` back to the module/formula documentation.
- Keep the wording scoped to documentation cross-references only.

## Test Plan
- `python3 -m py_compile torch/nn/modules/activation.py torch/_torch_docs.py torch/_tensor_docs.py torch/nn/functional.py torch/special/__init__.py`

### CODE REVIEW

**Comment by deepujain** [comment]
Not asking for review yet; I checked for rendered-doc evidence and only see the fork-safe checks in the visible rollup, not a `docs-build` artifact. The docs-preview generated pages for the sigmoid cross-reference docs are not available yet.

Could a maintainer approve/run docs-build when convenient? Once the preview or artifact exists, I can add rendered screenshots for the sigmoid API cross-references.
